packages/core/src/Stores/client-store.js

```diff
@@ -1957,16 +1957,20 @@ export default class ClientStore extends BaseStore {
     }
 
     setResidence(residence) {
-        this.accounts[this.loginid].residence = residence;
+        if (this.accounts[this.loginid]) {
+            this.accounts[this.loginid].residence = residence;
```

Comment:
can't we use optional chaining?
```suggestion
            this.accounts[this.loginid]?.residence = residence;
```

Reply:
optional chaning wont work here since we are assigning values here
<img width="752" alt="Screenshot 2023-04-11 at 7 51 22 PM" src="https://user-images.githubusercontent.com/84661147/231219155-7a788954-4dad-4f0b-b1c1-0c5ab33b1d6d.png">